sling.go

```diff
@@ -19,10 +19,14 @@ const (
 	formContentType = "application/x-www-form-urlencoded"
 )
 
+type Doer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 // Sling is an HTTP Request builder and sender.
 type Sling struct {
 	// http Client for doing requests
-	httpClient *http.Client
+	doer Doer
```

Comment:
let's just keep the unexported variable called `httpClient` or `client` since that still reflects its purpose when reading through